realm/src/main/java/io/realm/Realm.java

```diff
@@ -306,18 +310,15 @@ private static Realm createAndValidate(RealmConfiguration configuration, boolean
                     throw e;
                 }
             }
-            setupColumnIndices(realm);
+            final Map<Class<? extends RealmObject>, ColumnInfo> columnInfoMap = validatedRealmFiles.get(configuration.getPath());
```

Comment:
Any reason this is final? Also we could probably either remove the setupColumnIndices method or move this line down into it as well?